aea/launcher.py

```diff
@@ -0,0 +1,251 @@
+
+    :param agent_dir: agent configuration directory
+    :param stop_event: multithreading Event to stop agent run.
+
+    :return: None
+    """
+    agent = load_agent(agent_dir)
+
+    def stop_event_thread():
+        stop_event.wait()
+        agent.stop()
+
+    Thread(target=stop_event_thread, daemon=True).start()
+    try:
+        agent.start()
+    except Exception as e:
+        exc = AEAException(f"Raised {type(e)}({e})")
+        exc.__traceback__ = e.__traceback__
+        raise exc
+    finally:
+        stop_event.set()
+
+
+class ProcessExecutor(BaseAEAExecutor):
+    """
+    AEA executor uses subprocess to run multiple AEAs.
+
+    Support only AEA direcories, not instances.
+    """
+
+    def __init__(self, agent_dirs: Sequence[PathLike]) -> None:
+        """
+        Init executor.
+
+        :param agents: sequence of agents dirs to run.
+        """
+        super().__init__(agent_dirs)  # type: ignore # tricky part, cause need to pass str instead of AEA
+        self._agents: Sequence[PathLike] = agent_dirs  # type: ignore # tricky part, cause need to pass str instead of AEA
+        self._task_stop_event: Dict[Awaitable, Event] = {}
+
+    def _set_executor_pool(self):
+        """Set ProcessPoolExecutor to be used."""
+        self._executor_pool = ProcessPoolExecutor(max_workers=len(self._agents))
+        self._manager = multiprocessing.Manager()
+
+    def _start_agent(self, agent_dir: PathLike) -> Awaitable:  # type: ignore # tricky part, cause need to pass str instead of AEA
+        """
+        Start particular agent.
+
+        :param agent: agent dir with configuration to start.
+        :return: awaitable task to get result or exception
+        """
+        stop_event = self._manager.Event()
+        task = self._loop.run_in_executor(
+            self._executor_pool, _run_agent, agent_dir, stop_event
+        )
+        self._task_stop_event[task] = stop_event
+        return task
+
+    def _stop_agent(self, agent_dir: PathLike):  # type: ignore # tricky part, cause need to pass str instead of AEA
+        """
+        Stop particular agent.
+
+        :param agent: AEA instance to stop.
+        :return: None
+        """
+        task = self._agent_task[agent_dir]  # type: ignore # tricky part, cause need to pass str instead of AEA
+        stop_event = self._task_stop_event[task]
+        stop_event.set()
+
+
+class BaseLaunchExecutor:
+    """Base launch executor to implement executors."""
+
+    RUNNER_EXECUTOR_CLASS: Optional[Type[BaseAEAExecutor]] = None
+
+    def __init__(self, agent_dirs: List[PathLike]) -> None:
+        """
+        Inin launche executor.
+
+        :param agent_dirs: sequence of agents configuration directories.
+        """
+        self._agent_dirs = agent_dirs
+        self._runner: Optional[BaseAEAExecutor] = None
+        if not self.RUNNER_EXECUTOR_CLASS:
+            raise ValueError("RUNNER_EXECUTOR_CLASS has to be specified!")
```

Comment:
Ok, so why make it optional in the class var? why not set a default

Reply:
cause it's base class, without executor defined